tests/unit/core/mender-artifact_test.cpp

```diff
@@ -0,0 +1,314 @@
+
+    ASSERT_EQ(ctx, ctx2);
+    mender_artifact_release_ctx(ctx);
+}
+
+TEST_F(MenderArtifactTest, ProcessData) {
+    mender_artifact_ctx_t *ctx = mender_artifact_create_ctx(1024);
+
+    /* Process artifact data */
+    ASSERT_EQ(MENDER_OK, mender_artifact_process_data(ctx, data, artifact_data.size(), mock_download_data));
+
+    /* Number of payloads in the artifact, expect 1 */
+    ASSERT_EQ(ctx->payloads.size, 1);
+
+    /* Type of payload, expect `unit-test` */
+    mender_artifact_payload_t *payload = ctx->payloads.values;
+    ASSERT_STREQ(payload->type, "unit-test");
+
+    /* The original context size was too small, so the new one should be larger
+     * than the original isze */
+    ASSERT_GT(ctx->input.size, ctx->input.orig_size);
+
+    /* Should be 0/NULL, as there are no artifacts currently
+     * being parsed */
+    ASSERT_EQ(ctx->file.size, 0);
+    ASSERT_EQ(ctx->file.index, 0);
+    ASSERT_STREQ(ctx->file.name, nullptr);
+
+    /* One thing should be provided, the artifact name */
+    mender_key_value_list_t *provides = ctx->artifact_info.provides;
+    ASSERT_STREQ(provides->value, "unit-test-artifact");
+
+    /* Should depend on one thing, the artifact type */
+    mender_key_value_list_t *depends = ctx->artifact_info.depends;
+    ASSERT_STREQ(depends->value, "unit-test");
+
+    /* Check the stream state */
+    ASSERT_EQ(ctx->stream_state, MENDER_ARTIFACT_STREAM_STATE_PARSING_HEADER);
+
+    mender_artifact_release_ctx(ctx);
+}
+
+TEST_F(MenderArtifactTest, ProcessData_TooLargeSize) {
+    mender_artifact_ctx_t *ctx = mender_artifact_create_ctx(1024);
+
+    /* Assert that if file.size is too big, we realloc correctly */
+    ctx->file.size = 99999999999999999;
+    ASSERT_EQ(MENDER_OK, mender_artifact_process_data(ctx, data, artifact_data.size(), mock_download_data));
+
+    mender_artifact_release_ctx(ctx);
+
+    ctx = mender_artifact_create_ctx(1024);
+
+    /* If the input size is too big, we're not able to realloc, and should fail */
+    ctx->input.length = 99999999999999999;
+    ASSERT_EQ(MENDER_FAIL, mender_artifact_process_data(ctx, data, artifact_data.size(), mock_download_data));
+
+    mender_artifact_release_ctx(ctx);
+}
+
+TEST_F(MenderArtifactTest, ProcessData_EmptyInput) {
+    mender_artifact_ctx_t *ctx = mender_artifact_create_ctx(1024);
+
+    ASSERT_EQ(MENDER_OK, mender_artifact_process_data(ctx, nullptr, 0, mock_download_data));
+
+    ASSERT_EQ(ctx->input.size, ctx->input.orig_size);
+    ASSERT_EQ(ctx->file.size, 0);
+    ASSERT_EQ(ctx->file.index, 0);
+    ASSERT_STREQ(ctx->file.name, nullptr);
+
+    mender_artifact_release_ctx(ctx);
+}
+
+TEST_F(MenderArtifactTest, ProcessData_LargeInput) {
+    mender_artifact_ctx_t *ctx = mender_artifact_create_ctx(1024);
+    string                 large_data(10240, 'A');
+
+    ASSERT_EQ(MENDER_FAIL, mender_artifact_process_data(ctx, (void *)large_data.c_str(), large_data.size(), mock_download_data));
```

Comment:
Does this fail because of large data or because it's not valid artifact data?

Reply:
The name is indeed misleading; it has nothing to do with the size of the data